File: Day 1/1.py
with open("1.txt", "r") as f:
    i = [int(x) for x in f.read().splitlines()]
# Part 1
print(sum([x < y for (x, y) in zip(i[:-1], i[1:])]))
# Part 1
print(sum(([x < y for (x, y) in zip(i[:-3], i[3:])])))

# Part 2 alt
n = [x + y + z for (x, y, z) in zip(i, i[1:], i[2:])]
print(sum([x < y for (x, y) in zip(n[:-1], n[1:])]))

File: Day 1/1.hs
import System.IO

getInput :: IO [Int]
getInput = do f <- readFile "1.txt"
              return $ map (read :: String -> Int) (lines f)

eval :: Int -> Int -> Int -> Int
eval acc a b = acc + (fromEnum $ b > a)

solve :: [Int] -> Int -> Int -> Int
solve ls n acc
    | length ls < (n+1) = acc
    | otherwise = solve (tail ls) (n) $ eval (acc) (head ls) (ls !! n)

main :: IO()
main = do
    i <- getInput
    -- Part 1
    print $ solve i 1 0
    -- Part 2
    print $ solve i 3 0
